services/autorust/codegen/src/codegen_operations.rs

```diff
@@ -329,6 +328,16 @@ impl WebOperationGen {
             next_link_name: p.next_link_name.clone(),
         })
     }
+
+    pub fn success_responses(&self) -> IndexMap<StatusCode, Response> {
+        let mut map = IndexMap::new();
+        for (status_code, rsp) in &self.0.responses {
+            if crate::status_codes::is_success(status_code) {
+                map.insert(status_code.to_owned(), rsp.to_owned());
+            }
+        }
+        map
```

Comment:
```suggestion
       self.0.responses.iter().cloned().filter(|(status_code, _) crate::status_codes::is_success(status_code)).collect()
```

Reply:
I couldn't make this compile, but I applied the rest.